gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/BaselineJavaVersion.java

```diff
@@ -71,15 +72,15 @@ public void execute(JavaToolchainSpec javaToolchainSpec) {
             configureExecutionTasks(project, javaToolchains.forVersion(extension.runtime()));
 
             // Validation
-            project.getTasks()
+            TaskProvider<CheckJavaVersionsTask> checkJavaVersions = project.getTasks()
                     .register("checkJavaVersions", CheckJavaVersionsTask.class, new Action<CheckJavaVersionsTask>() {
                         @Override
                         public void execute(CheckJavaVersionsTask task) {
                             task.getTargetVersion().set(extension.target());
                             task.getRuntimeVersion().set(extension.runtime());
-                            project.getTasks().getByName("check").dependsOn(task);
```

Comment:
This is a problem because check would only depend on checkJavaVersions if checkJavaVersions was configured.